desktop/ui/src/main/java/org/datacleaner/windows/AbstractDatastoreDialog.java

```diff
@@ -120,16 +120,20 @@ public void actionPerformed(ActionEvent e) {
         _datastoreNameTextField.getDocument().addDocumentListener(new DCDocumentListener() {
             @Override
             protected void onChange(DocumentEvent event) {
-                validateAndUpdate();
+                validateAndUpdateInternal();
             }
         });
     }
-    
+
     protected void validateAndUpdate() {
+        validateAndUpdateInternal();
+    }
+
+    private void validateAndUpdateInternal() {
```

Comment:
Does this change anything? Wouldn't subclasses just override validateForm()? Or maybe that should only be really for the form inputs individually. Maybe we should document what subclasses should/could override to make it clearer.